### PR TITLE
WA-4414: SDK manual transaction create update delete API

### PR DESCRIPTION
**Pull request by hailefrollo**
Checklist
Use the ✅ emoji for when it's been done; ⛔ when it still needs work; strikethrough when an item of the checklist does not apply to the changes.

- Is the destination branch correct & intended? ✅
- Is the correct JIRA ticket referenced? ✅
- Is the correct API docs/Confluence Page referenced? ✅
- Have all the AC's in the ticket passed? ✅
- Have you followed proper naming conventions in the code? ✅
- Has the code been properly covered with unit tests?✅
- Have you regenerated the mock DBs, if the datamodel was modified✅
- Have you added migration tests (if applicable)?✅
- Have the unit tests passed locally (if applicable)?✅
- Have you addressed the code warnings if any?✅
- Have you tested the API and made sure the documentation is up-to-date?✅
- Have you added documentation for all public methods & types✅

### CODE REVIEW

File: Sources/FrolloSDK/Aggregation/Aggregation.swift [hunk 1]
```diff
@@ -3837,4 +3925,57 @@ public class Aggregation: CachedObjects, ResponseHandler {
...
+                                 objectResponse: response,
+                                 primaryKey: #keyPath(Transaction.transactionID),
+                                 managedObjectContext: managedObjectContext)
+
+        managedObjectContext.performAndWait {
+            do {
+                try managedObjectContext.save()
+
+                // is this the correct place to do this ?
+                NotificationCenter.default.post(name: Aggregation.refreshTransactionsNotification,
```
Comment (sincychacko-frollo) [inline Sources/FrolloSDK/Aggregation/Aggregation.swift:3948]: Fine here. But let's use transactionsUpdatedNotification instead

File: Sources/FrolloSDK/Model/API/Aggregation/APIManualTransactionCreateUpdateRequest.swift [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    
+    /// Direct debit
+    case directDebit = "direct_debit"
+    
+    /// Other
+    case other
+}
+
+/// Data representation of the Manual Transaction Create/Update Request
+public struct APIManualTransactionCreateUpdateRequest: Codable {
```
Comment (sincychacko-frollo) [inline Sources/FrolloSDK/Model/API/Aggregation/APIManualTransactionCreateUpdateRequest.swift:48]: doesn't need to be public

File: Sources/FrolloSDK/Resources/Database/FrolloSDKDataModel.xcdatamodeld/FrolloSDKDataModel-1.8.2.xcdatamodel/contents
Comment (sincychacko-frollo) [inline Sources/FrolloSDK/Resources/Database/FrolloSDKDataModel.xcdatamodeld/FrolloSDKDataModel-1.8.2.xcdatamodel/contents:1]: Let's add a new model version as it's going in the next release

Comment (hailefrollo) [reply Sources/FrolloSDK/Resources/Database/FrolloSDKDataModel.xcdatamodeld/FrolloSDKDataModel-1.8.2.xcdatamodel/contents:1]: @sincychacko-frollo thanks, I updated my PR

File: Sources/FrolloSDK/Aggregation/Aggregation.swift [hunk 3]
```diff
@@ -3837,4 +4009,56 @@ public class Aggregation: CachedObjects, ResponseHandler {
...
+        removeObject(type: Transaction.self,
+                     id: transactionID,
+                     primaryKey: #keyPath(Transaction.transactionID),
+                     managedObjectContext: managedObjectContext)
+
+        managedObjectContext.performAndWait {
+            do {
+                try managedObjectContext.save()
+                // is this the correct place to do this ?
+                NotificationCenter.default.post(name: Aggregation.refreshTransactionsNotification,
```
Comment (sincychacko-frollo) [inline Sources/FrolloSDK/Aggregation/Aggregation.swift:4057]: Can we update here as well.

Comment (hailefrollo) [reply Sources/FrolloSDK/Aggregation/Aggregation.swift]: Thanks, I updated my PR!

**Review by sincychacko-frollo — APPROVED** [review]

**Pull request merged**